contracts/SwapKiwiV2.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.1;
+
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract SwapKiwiV2 is Ownable, ERC721Holder, ERC1155Holder {
+
+    uint256 private _swapsCounter;
+    uint256 private _etherLocked;
+
+    uint256 public fee;
+
+    mapping (uint256 => Swap) private _swaps;
+
+    struct Swap {
+		address payable initiator;
+		address[] initiatorNftAddresses;
+		uint256[] initiatorNftIds;
+		uint256[] initiatorNftAmounts;
+		uint256 initiatorEtherValue;
```

Comment:
we could use uint128 for initiatorEtherValue and secondUserEtherValue and reorder them together so they can be packed

Reply:
make sense.